sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -347,6 +347,15 @@ querySpecification
        windows?)
     ;
 
+hint
+    : '/*+' hintStatement '*/'
+    ;
+
+hintStatement
+    : hintName=identifier identifierList
+    | hintName=identifier '(' parameter1=identifier parameter2=identifier ')'
```

Comment:
I think you can also do this (is easier in the AST builder): `| hintName=identifier '(' parameters+=identifier parameters+=identifier ')'`